spec/services/course/survey/reminder_service_spec.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+RSpec.describe Course::Survey::ReminderService do
+  let(:instance) { Instance.default }
+  with_tenant(:instance) do
+    let(:course) { create(:course) }
+    let(:survey) do
+      create(:survey, course: course, published: true, end_at: Time.zone.now + 1.day,
+                      creator: course.creator, updater: course.creator)
+    end
+    let!(:unresponded_student) { create(:course_student, course: course) }
+    let!(:responded_student) { create(:course_student, course: course) }
+    let!(:response) do
+      create(:response, survey: survey, course_user: responded_student,
+                        submitted_at: Time.zone.now,
+                        creator: responded_student.user, updater: responded_student.user)
+    end
+    let(:course_creator_email) { course.creator.email }
+    let(:unresponded_student_email) { unresponded_student.user.email }
+    let(:responded_student_email) { responded_student.user.email }
+
+    describe '#opening_reminder' do
+      subject do
+        Course::Survey::ReminderService.opening_reminder(survey, survey.opening_reminder_token)
+      end
+
+      it 'notifies all course users' do
+        subject
+        emails = ActionMailer::Base.deliveries.map(&:to).map(&:first)
+        expect(emails).to include(course_creator_email)
+        expect(emails).to include(unresponded_student_email)
+        expect(emails).to include(responded_student_email)
+
+        expect(ActionMailer::Base.deliveries.first.body.parts.first.body).
```

Comment:
This might not be threadsafe cause other tests could also write to `ActionMailer::Base.deliveries`,  and then `first` could not be the email you want. 

possible to the `emails` defined above ?